http_parser.c

```diff
@@ -1782,9 +1782,9 @@ size_t http_parser_execute (http_parser *parser,
 
         if (parser->flags & F_TRAILING) {
           /* End of a chunked request */
-          UPDATE_STATE(NEW_MESSAGE());
-          CALLBACK_NOTIFY(message_complete);
-          break;
+          UPDATE_STATE(s_message_done);
```

Comment:
Well, the previous code has quite similar behaviour, except that the state will be different in chunk_complete_cb. I just want to be sure I understand the reasons behind this. Is it possible to ask the person who implemented it?